backend/controller/user.go

```diff
@@ -1,87 +1,33 @@
 package controller
 
 import (
-	"api/config"
 	"api/helpers"
 	"api/models"
 	"encoding/json"
 	"github.com/go-chi/chi/v5"
-	"log"
 	"net/http"
 )
 
-func GetAllUsers(w http.ResponseWriter, r *http.Request) {
+func GetAllUsers(w http.ResponseWriter, _ *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	users := []models.User{}
-	result := config.DB.Find(&users)
-	if result.Error != nil {
-		helpers.DBErrorHandling(result.Error, w, r)
-		return
-	}
-	//go yells at you if you don't error check this, better to be safe
-	errJson := json.NewEncoder(w).Encode(&users)
-	if errJson != nil {
-		http.Error(w, "500", http.StatusInternalServerError)
-		return
-	}
+	helpers.GetAllTable(w, []models.User{})
 }
 func GetUser(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	uId := chi.URLParam(r, "userId")
-	var user models.User
-	result := config.DB.First(&user, uId)
-
-	if result.Error != nil {
-		helpers.DBErrorHandling(result.Error, w, r)
-		return
-	}
-	errJson := json.NewEncoder(w).Encode(&user)
-	if errJson != nil {
-		http.Error(w, "500", http.StatusInternalServerError)
-		return
-	}
+	helpers.GetById(w, models.User{}, uId)
 }
 
 func CreateUser(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	var user models.UserReq
+	var user models.User
 	err := json.NewDecoder(r.Body).Decode(&user)
-	if err != nil || !helpers.CheckFields(user) {
-		if err != nil {
-			log.Print(err)
-		} else {
-			log.Print("Request doesn't contain all params")
-		}
-		http.Error(w, "400", http.StatusBadRequest)
-		return
-	}
-	dbUser := models.User{Name: user.Name,
-		Email: user.Email, IsAdmin: user.IsAdmin}
-	result := config.DB.Create(&dbUser)
-	if result.Error != nil {
-		helpers.DBErrorHandling(result.Error, w, r)
-		return
-	}
-	errJson := json.NewEncoder(w).Encode(dbUser)
-	if errJson != nil {
-		http.Error(w, "500", http.StatusInternalServerError)
-		return
-	}
-
+	user.Common = models.Common{}
```

Comment:
essentially ignore any inputs the user made to anything within "common", which should not be writeable